src/util/compatibility.h

```diff
@@ -157,30 +157,30 @@ inline T* atomicLoadRelaxed(QAtomicPointer<T> atomicPtr) {
 }
 
 template <typename T>
-inline void atomicStoreRelaxed(QAtomicInteger<T> atomicInt, T newValue) {
+inline void atomicStoreRelaxed(QAtomicInteger<T>* atomicInt, T newValue) {
     // TODO: QBasicAtomicInteger<T>::store(T newValue) is deprecated and should
     // be replaced with QBasicAtomicInteger<T>::storeRelaxed(T newValue)
     // However, the proposed alternative has just been introduced in Qt 5.14.
     // Until the minimum required Qt version of Mixx is increased, we need a
     // version check here
 #if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
-    atomicInt.storeRelaxed(newValue);
+    atomicInt->storeRelaxed(newValue);
 #else
-    atomicInt.store(newValue);
+    atomicInt->store(newValue);
 #endif
 }
 
 template <typename T>
-inline void atomicStoreRelaxed(QAtomicPointer<T> atomicPtr, T* newValue) {
+inline void atomicStoreRelaxed(QAtomicPointer<T>* atomicPtr, T* newValue) {
```

Comment:
For those compatibility functions passing a non-const ref would IMHO be more appropriate to preserve the symmetry with the corresponding const load functions.

We are mimicking member functions and should pass either a reference or a pointer, but not both depending on the signature.

Reply:
That was my initial fix, but @daschuer said I shouldn't use non-const references: https://github.com/mixxxdj/mixxx/pull/2423#discussion_r373712931

Reply:
I disagree for this special case. Blindly applying some custom rules that then cause other inconsistencies is not helpful.

Reply:
As a general rule of thumb, it is ok to break general style rules in the case of internal libraries / compatibility layers / adapters like this, especially if it makes the API easier to use and calling code more easy to adhere to those styles.  

The reason not to use a non-const ref is that it can be hard to differentiate a ref from a local and this can cause crash bugs, so we do have to be careful. But in this case I agree that it's better to keep the API consistent for callers.

Reply:
OK, I restored the original commit.